pkg/capabilities/pb/capabilities_helpers.go

```diff
@@ -100,8 +105,13 @@ func CapabilityResponseFromProto(pr *CapabilityResponse) capabilities.Capability
 	if pr.Error != "" {
 		err = errors.New(pr.Error)
 	}
-	return capabilities.CapabilityResponse{
-		Value: val,
-		Err:   err,
+
+	resp := capabilities.CapabilityResponse{
+		Err: err,
 	}
+	if mval, ok := val.(*values.Map); ok {
```

Comment:
1. when would this ever not be the case?
2. should it be an error if !ok ? if no, why not and should it at least be logged?

Reply:
The value can be nil which would be a different type of value type. It doesn't need to be an error if !ok because the only way to construct a CapabilityResponse proto is via the helper above: https://github.com/smartcontractkit/chainlink-common/pull/540/files/0f8e143ef729e1e50eb13e527e5b96b5b2711602#diff-14830fdbb3d14caa2427e5d7032446d63c5f6ae4a50df50414e601c2720c74ecR79

Reply:
i read the values library and i sort of understand. i think my confusion boils down to Wrap(nil) => nil
and so `nil` is used on both side of the dynamic type system and it means different things on each side.

Reply:
Yes, I agree that the handling of nil is a bit confusing: I initially had a message type called values.Nil which represented a nil value, but that was actually a terrible value since that meant we had to handle nil and values.Nil correctly.

Reply:
that makes sense. i was looking at the library and thinking about having an Nil type, and i got stuck about what to do with nil of the interface instance itself